src/participant/prompts/query.ts

```diff
@@ -38,22 +48,32 @@ db.getCollection('').find({
   date: { $gte: new Date('2014-04-04'), $lt: new Date('2014-04-05') }
 }).count();
 
-Database name: ${databaseName}
-Collection name: ${collectionName}
-${
-  schema
-    ? `Collection schema:
-${schema}`
-    : ''
-}
-
 MongoDB command to specify database:
 use('');
 
 MongoDB command to specify collection:
-db.getCollection('')
-
-Concisely explain the code snippet you have generated.`;
+db.getCollection('');\n\n`;
+    if (databaseName) {
+      prompt += `Database name: ${databaseName}\n`;
+    }
+    if (collectionName) {
+      prompt += `Collection name: ${collectionName}\n`;
+    }
+    if (schema) {
+      prompt += `Collection schema: ${schema}\n`;
+    }
+    if (sampleDocuments) {
+      let docs = toJSString(sampleDocuments);
+      // First check the length of all stringified sample documents.
+      // If the resulting prompt is too large, proceed with only 1 sample document.
+      if (prompt.length + docs.length > maxInputTokens) {
```

Comment:
There's a difference in string length and token counts. Token count is the numerical result of running a tokenizer on the input string. We can get it by running the `countTokens` method on the `LanguageModelChat` https://code.visualstudio.com/api/references/vscode-api#LanguageModelChat.
Can we update this and the check below to count the tokens of the total prompt length?

Reply:
If this also has the restriction on the model first run that this method isn't available let's do a similar thing for the first question asked, where we default it to a set variable, if that sounds good to you.
Mostly thinking about how we can avoid setting a limit on our end while the model's context size does change overtime.